snowflake/src/test/java/org/apache/iceberg/snowflake/SnowflakeCatalogTest.java

```diff
@@ -0,0 +1,231 @@
+    // Existence check for nonexistent parent namespaces is optional in the SupportsNamespaces
+    // interface.
+    String dbName = "NONEXISTENT_DB";
+    Assert.assertThrows(RuntimeException.class, () -> catalog.listNamespaces(Namespace.of(dbName)));
+  }
+
+  @Test
+  public void testListNamespaceWithinSchema() {
+    // No "sub-namespaces" beyond database.schema; invalid to try to list namespaces given
+    // a database.schema.
+    String dbName = "DB_3";
+    String schemaName = "SCHEMA_4";
+    Assert.assertThrows(
+        IllegalArgumentException.class,
+        () -> catalog.listNamespaces(Namespace.of(dbName, schemaName)));
+  }
+
+  @Test
+  public void testListTables() {
+    List<TableIdentifier> tables = catalog.listTables(Namespace.empty());
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_1"),
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_2"),
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_3"),
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_4"),
+            TableIdentifier.of("DB_3", "SCHEMA_3", "TAB_5"),
+            TableIdentifier.of("DB_3", "SCHEMA_4", "TAB_6")),
+        tables);
+  }
+
+  @Test
+  public void testListTablesWithinDB() {
+    String dbName = "DB_1";
+    List<TableIdentifier> tables = catalog.listTables(Namespace.of(dbName));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_1"),
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_2")),
+        tables);
+  }
+
+  @Test
+  public void testListTablesWithinNonexistentDB() {
+    String dbName = "NONEXISTENT_DB";
+    Assert.assertThrows(RuntimeException.class, () -> catalog.listTables(Namespace.of(dbName)));
+  }
+
+  @Test
+  public void testListTablesWithinSchema() {
+    String dbName = "DB_2";
+    String schemaName = "SCHEMA_2";
+    List<TableIdentifier> tables = catalog.listTables(Namespace.of(dbName, schemaName));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_3"),
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_4")),
+        tables);
+  }
+
+  @Test
+  public void testListTablesWithinNonexistentSchema() {
+    String dbName = "DB_2";
+    String schemaName = "NONEXISTENT_DB";
+    Assert.assertThrows(
+        RuntimeException.class, () -> catalog.listTables(Namespace.of(dbName, schemaName)));
+  }
+
+  @Test
+  public void testLoadS3Table() {
+    Table table = catalog.loadTable(TableIdentifier.of(Namespace.of("DB_1", "SCHEMA_1"), "TAB_1"));
+    Assert.assertEquals(table.location(), "s3://tab1/");
+  }
+
+  @Test
+  public void testLoadAzureTable() {
+    Table table = catalog.loadTable(TableIdentifier.of(Namespace.of("DB_2", "SCHEMA_2"), "TAB_3"));
+    Assert.assertEquals(
+        table.location(), "wasbs://mycontainer@myaccount.blob.core.windows.net/tab1/");
```

Comment:
`assertEquals(expected, actual)` is the signature, so I think the params should be turned around :)
However, it would be great to switch this (and potentially other places above) to `Assertions.assertThat(table.location().isEqualTo(...)` as that is more fluent to read

Reply:
Done, switched to Assertions.assertThat